ml-agents-envs/mlagents_envs/env_utils.py

```diff
@@ -106,3 +107,17 @@ def launch_executable(file_name: str, args: List[str]) -> subprocess.Popen:
                 f"permissions are set correctly. For example "
                 f'"chmod -R 755 {launch_string}"'
             ) from perm
+
+
+def returncode_to_signal_name(returncode: int) -> Optional[str]:
+    """
+    Try to convert return codes into their corresponding signal name.
+    E.g. returncode_to_signal_name(-2) -> "SIGINT"
+    """
+    try:
+        # A negative value -N indicates that the child was terminated by signal N (POSIX only).
+        s = signal.Signals(-returncode)  # pylint: disable=no-member
```

Comment:
If the return code from the Unity exe was positive, it might not be a signal.  So now an application error return code `2` would be interpreted as `SIGINT` which isn't what you want I think

Reply:
(this is just migrated code)
Guess I should have cited my source - that comment is from https://python.readthedocs.io/en/latest/library/subprocess.html#subprocess.CompletedProcess.returncode

If the application returns 2, we'll try to compute signal.Signals(-2) which will raise since it's not a valid enum value (and we catch below):
```
>>> import signal
>>> signal.Signals(-2)
ValueError: -2 is not a valid Signals
```

Reply:
Oh, I see!  It's expecting the positive version of the signal 😂 , the API is just confusing.